### ISSUE TITLE
Shadowing Doesn't work

### ISSUE THREAD

**Issue by mrgamer45** [JEFF-Media-GbR/CustomBlockData#5]
Not sure if this is from the library itself or just my bad maven skills, but pasting all the shadowing/repo/dependency stuff and changing it doesn't seem to want to actually shadow. In IntelliJ, the IDE recognizes the dependency, but after building, I get the NoClassDefFound error. My guess is its just an IntelliJ setting, but still making the issue just in case

**Comment by mfnalex** [comment]
Show your pom.xml please. You have to add the maven-shade-plugin and also add the proper execution goal/phase.

**Comment by mrgamer45** [comment]
Oh I forgot the executions part

**Comment by mfnalex** [comment]
yes that happens quite often. I should update the readme file to include the full maven-shade part :)

**Comment by mrgamer45** [comment]
I just realized I've been saying shadowing when I made this issue

**Comment by mfnalex** [comment]
Oh btw you mentioned this:

> My guess is its just an IntelliJ setting, but still making the issue just in case

Be sure that you also use maven to build the project (`mvn package`) and NOT use IntelliJ's build feature. Otherwise, shading will still not work.

**Comment by mrgamer45** [comment]
I feel like they must've changed the build thing, because I've used the exact same shading on different plugins and build worked but nowadays it doesnt

**Comment by mfnalex** [comment]
![image](https://user-images.githubusercontent.com/1122571/165373422-f17abccf-9815-4d99-b5f5-d47a69e6bbe2.png)
Just double click on "package" and maven will build and shade everything as defined in your pom.xml

**Comment by mfnalex** [comment]
If you don't get it to work, feel free to join my Discord and open a ticket, it's probably easier than "chatting" in a github issue lol
https://discord.jeff-media.com

**Comment by mrgamer45** [comment]
Ok

**Comment by mrgamer45** [comment]
> ![image](https://user-images.githubusercontent.com/1122571/165373422-f17abccf-9815-4d99-b5f5-d47a69e6bbe2.png) Just double click on "package" and maven will build and shade everything as defined in your pom.xml

how would I make it package to a different folder, like my plugins folder (I'm lazy lol) like how the build does it

**Comment by mfnalex** [comment]
https://blog.jeff-media.com/common-maven-questions/

**Comment by mrgamer45** [comment]
thx
